sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/service/classloader/AbstractClasspathClassloader.java

```diff
@@ -0,0 +1,253 @@
+ */
+package com.alipay.sofa.ark.container.service.classloader;
+
+import com.alipay.sofa.ark.common.util.StringUtils;
+import com.alipay.sofa.ark.container.service.ArkServiceContainerHolder;
+import com.alipay.sofa.ark.exception.ArkLoaderException;
+import com.alipay.sofa.ark.loader.jar.Handler;
+import com.alipay.sofa.ark.spi.service.classloader.ClassloaderService;
+import sun.misc.CompoundEnumeration;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+
+/**
+ *
+ * Abstract Classpath Classloader, basic logic to load class/resource, sub class need to implement
+ *
+ * @author ruoshan
+ * @since 0.1.0
+ */
+public abstract class AbstractClasspathClassloader extends URLClassLoader {
+
+    protected static final String CLASS_RESOURCE_SUFFIX = ".class";
+
+    protected ClassloaderService  classloaderService    = ArkServiceContainerHolder.getContainer()
+                                                            .getService(ClassloaderService.class);
+
+    public AbstractClasspathClassloader(URL[] urls) {
+        super(urls, null);
+    }
+
+    @Override
+    protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+        if (StringUtils.isEmpty(name)) {
+            return null;
+        }
+        Handler.setUseFastConnectionExceptions(true);
+        try {
+            return loadClassInternal(name, resolve);
+        } finally {
+            Handler.setUseFastConnectionExceptions(false);
+        }
+    }
+
+    /**
+     * Real logic to load class，need to implement by Sub Classloader
+     * @param name
+     * @param resolve
+     * @return
+     * @throws ArkLoaderException
+     */
+    abstract protected Class<?> loadClassInternal(String name, boolean resolve)
+                                                                               throws ArkLoaderException;
+
+    @Override
+    public URL getResource(String name) {
+        Handler.setUseFastConnectionExceptions(true);
+        try {
+            URL url;
+            // first find import resource
+            if (shouldFindExportedResource(name)) {
+                List<ClassLoader> exportResourceClassloadersInOrder = classloaderService
+                    .findExportResourceClassloadersInOrder(name);
+                if (exportResourceClassloadersInOrder != null) {
+                    for (ClassLoader exportResourceClassloader : exportResourceClassloadersInOrder) {
+                        url = exportResourceClassloader.getResource(name);
+                        if (url != null) {
+                            return url;
+                        }
+                    }
+                }
+
+            }
+
+            url = super.getResource(name);
+
+            if (url != null) {
+                return url;
+            }
+
+            if (name.endsWith(CLASS_RESOURCE_SUFFIX)) {
```

Comment:
when find class file resource, maybe it should prefer to find exported resource to local resource?